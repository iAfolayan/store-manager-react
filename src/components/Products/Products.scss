.bg{
  background-color: #00478e;
}
.text-primary {
  color: #00478e !important
}