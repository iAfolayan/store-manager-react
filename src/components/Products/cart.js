import React, { Component } from 'react';
import { connect } from 'react-redux';
import { removeFromCart } from '../../actions/productsActions';

export class cart extends Component {
	// calculateTotal() {
	// 	const total = cartItems.reduce(
	// 		(curr, next) => {
	// 			curr.quantity += next.quantity;
	// 			curr.amount += next.price * next.quantity;
	// 			return curr;
	// 		},
	// 		{ amount: 0, quantity: 0 }
	// 	);
	// }

	remove(cartIndex) {
		this.props.remove(cartIndex);
	}

	showInfo(cart) {
		if (cart.length) return '';
		else return 'hide';
	}

	render() {
		const { cart } = this.props.products;
		return (
			<div id='cart' className={`cart ${this.showInfo(cart)}`}>
				<span className='cartHeading'>Cart Info</span>
				<hr />
				{cart.map((each, index) => (
					<ul className='cartItemsList' key={index}>
						<img src={each.image} className='img-thumbnail' id='item-img' alt='' />
						<div className='item-text'>
							<p id='cart-item-title' className='cart-item-title'>
								{each.name}
							</p>
							<span>&#8358;</span>
							<span id='item-price' className='item-price'>
								{each.price}.00 -- {each.quantity}
							</span>
							<button onClick={() => this.remove(index)}>Remove</button>
						</div>
					</ul>
				))}
				{
					<div className='cartTotal'>
						Total: &#8358;<span id='amount'>0</span>
					</div>
				}
				<a href='#'>
					<button type='button' className='cartBtn'>
						Proceed to checkout
					</button>
				</a>
			</div>
		);
	}
}

const mapStateToProps = (state) => ({
	products: state.productsReducer
});

const mapDispatchToProps = (dispatch) => ({
	remove: (cartIndex) => dispatch(removeFromCart(cartIndex))
});

export default connect(mapStateToProps, mapDispatchToProps)(cart);
