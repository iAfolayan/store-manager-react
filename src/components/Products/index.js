import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { fetchProducts} from '../../actions/productsActions';
import ProductList from '../ProductList';
import './Products.scss';

class Products extends Component {
	constructor() {
		super();
	}
	// handleChange = (event) => {
	// 	this.setState({ filterproduct: event.target.value });
	// };
	
	componentDidMount(){
		this.props.fetchProducts()
	}
	
	render() {
		let { products } = this.props.products;
		const { filterQuery } = this.props;
		products = filterQuery ?
								products.filter((product) => {
									return product.productname.toLowerCase().includes(filterQuery.toLowerCase());
								})
								: products;
		
		if (products.length === 0) {
			return <div className="text-danger text-center"><h3>No Product found...</h3></div>
		} else {
			return (
				<Fragment>
					<h5 className="justify-content-start text-primary ml-3">Available Products</h5>
					<ProductList products={products} />
				</Fragment>
			);
		}
	}
}


const mapDispatchToProps = dispatch => ({
	fetchProducts: () => dispatch(fetchProducts())
})


const mapStateToProps = state => ({
  products: state.productsReducer
});

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(Products);
