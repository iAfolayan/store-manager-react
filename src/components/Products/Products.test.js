import React from 'react';
import {shallow} from 'enzyme';
import Products from '.';

describe('<Products />', () => {
  test('should render product display without crashed', () => {
    const wrapper = shallow(<Products />);
    expect(wrapper).toMatchSnapshot();
  });
});