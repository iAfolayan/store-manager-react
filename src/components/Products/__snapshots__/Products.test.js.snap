// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Products /> should render product display without crashed 1`] = `
<Fragment>
  <h5
    className="justify-content-start text-primary ml-3"
  >
    Available Products
  </h5>
  <ProductList
    filter={
      Array [
        Object {
          "description": "A new brand product",
          "id": 1,
          "image": "test-file-stub",
          "minimumallowed": 40,
          "name": "Bluetooth",
          "price": 302.6,
        },
        Object {
          "description": "A new brand product",
          "id": 2,
          "image": "test-file-stub",
          "minimumallowed": 50,
          "name": "Bluetooth",
          "price": 302,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 3,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Samsung Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 4,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Tecno Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is aimed in recruiting only the top 1% of software talent in Africa with a 10 years mission of training 100, 000 developers in Africa. This is one of the company primary strengths, and this attracted investments from great people like Zuckerberg.",
          "id": 5,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Nokia Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 6,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Hawei Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 7,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Hawei Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 8,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Hawei Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 9,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Hawei Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 10,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Hawei Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 11,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Hawei Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 12,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Hawei Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 13,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Hawei Mobile",
          "price": 50000,
        },
      ]
    }
    products={
      Array [
        Object {
          "description": "A new brand product",
          "id": 1,
          "image": "test-file-stub",
          "minimumallowed": 40,
          "name": "Bluetooth",
          "price": 302.6,
        },
        Object {
          "description": "A new brand product",
          "id": 2,
          "image": "test-file-stub",
          "minimumallowed": 50,
          "name": "Bluetooth",
          "price": 302,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 3,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Samsung Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 4,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Tecno Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is aimed in recruiting only the top 1% of software talent in Africa with a 10 years mission of training 100, 000 developers in Africa. This is one of the company primary strengths, and this attracted investments from great people like Zuckerberg.",
          "id": 5,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Nokia Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 6,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Hawei Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 7,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Hawei Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 8,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Hawei Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 9,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Hawei Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 10,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Hawei Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 11,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Hawei Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 12,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Hawei Mobile",
          "price": 50000,
        },
        Object {
          "description": "Andela is a joint company founded in Nigeria and the U.S with a different location in Lagos, Kampala, Nairobi, Cairo, and Kigali",
          "id": 13,
          "image": "test-file-stub",
          "minimumallowed": 20,
          "name": "Hawei Mobile",
          "price": 50000,
        },
      ]
    }
  />
</Fragment>
`;
